SIPS/sip-28.md

```diff
@@ -0,0 +1,97 @@
+---
+sip: 28
+title: Background Events
+status: Draft
+author: Olaf Tomalka (@ritave)
+created: 2024-10-09
+---
+
+## Abstract
+
+This SIP introduces a way the schedule non-recurring events in the future.
+
+## Motivation
+
+Scheduled recurring events are already support in Snaps through Cronjobs feature. Introducing non-recurring events we allow novel use-cases for Snap developers, such as allowing a snap that sets a reminder for ENS domain expiration date.
+
+## Specification
+
+> Indented sections like this are considered non-normative.
+
+### Language
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT",
+"SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and
+"OPTIONAL" written in uppercase in this document are to be interpreted as described in [RFC 2119](https://www.ietf.org/rfc/rfc2119.txt)
+
+### Snap Manifest
+
+This SIP introduces a new permission `endowment:background-events`. This permissions grants a Snap the ability to schedule future events.
```

Comment:
Wondering if we should somehow merge this with cronjobs 🤔 Otherwise, we'll have two permissions that do extremely similar things 😄

Reply:
WDYT @Mrtenz?

Reply:
I think we eventually want to rename cronjobs to background events, but doing so will be a breaking change.